### PR TITLE
Homework end with appropriate message and review on past due

### PR DESCRIPTION
**Pull request by pandafulmanda**
screen shot 2015-04-27 at 3 44 10 pm
screen shot 2015-04-27 at 3 45 46 pm

- [x] tests
  - [x] can view review page and review page has expected complete and incomplete steps

### CODE REVIEW

**Comment by philschatz** [comment]
oh cool! Is there a mockup for this?

**Comment by Fredasaurus** [comment]
I will make one.

File: src/components/task-step/ends.cjsx [hunk 1]
```diff
@@ -26,16 +30,86 @@ PracticeEnd = React.createClass
...
+
+  _removeListener: ->
+    TaskStepStore.removeChangeListener(@_update)
+
+  componentWillMount:   -> @_addListener()
+  componentWillUnmount: -> @_removeListener()
+  componentWillUpdate: -> @_removeListener()
+  componentDidUpdate:  -> @_addListener()
+  # override focus from within task steps to focus on first to do question
+  componentDidMount: -> document.querySelector('.task-step textarea')?.focus()
```
Comment (philschatz) [inline src/components/task-step/ends.cjsx]: Can you use `@refs.foo.getDOMNode().focus()` instead?

Comment (pandafulmanda) [inline src/components/task-step/ends.cjsx]: since it's wrapped inside a loadable, it's a little tougher

Comment (philschatz) [inline src/components/task-step/ends.cjsx]: Why not just have the component that creates the `<textarea>` do the focus?

Comment (pandafulmanda) [inline src/components/task-step/ends.cjsx]: that is happening, on each of the component's mount.  if i leave this line out right now, it'll focus on the last textarea that mounts, the last one that appears on the page

File: src/components/task-step/ends.cjsx [hunk 2]
```diff
@@ -26,22 +36,96 @@ PracticeEnd = React.createClass
...
+      />
+
+    stepsReview =
+      <div className="task task-review-#{type}">
+        {label}
+        {stepsList}
+      </div>
+
+  renderAfterDue: (taskId) ->
+    completedSteps = TaskStore.getCompletedSteps taskId
```
Comment (philschatz) [inline src/components/task-step/ends.cjsx]: Can you include parens around 1-line function calls? I noticed some parts of this PR use parens and other parts don't

Comment (pandafulmanda) [inline src/components/task-step/ends.cjsx]: okay :smile:

Comment (philschatz) [inline src/components/task-step/ends.cjsx]: The maps aren't 1-liners. but yeah, we should have simpler code style rules. Maybe meet after the demo and discuss?

Comment (pandafulmanda) [inline src/components/task-step/ends.cjsx]: :+1:

**Pull request merged**